kaldb/src/test/java/com/slack/kaldb/logstore/opensearch/FieldLimitTest.java

```diff
@@ -0,0 +1,66 @@
+package com.slack.kaldb.logstore.opensearch;
+
+import static com.slack.kaldb.logstore.opensearch.OpenSearchAdapter.tryRegisterField;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.slack.kaldb.metadata.schema.FieldType;
+import com.slack.kaldb.metadata.schema.LuceneFieldDef;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.Test;
+
+public class FieldLimitTest {
+
+  @Before
+  public void resetTotalFieldsLimit() {
+    OpenSearchAdapter.TOTAL_FIELDS_LIMIT = 5000;
+  }
+
+  @AfterClass
+  public static void resetTotalFieldsLimitAfterTestRuns() {
+    OpenSearchAdapter.TOTAL_FIELDS_LIMIT = 5000;
+  }
+
+  @Test
+  public void checkIfMaxFieldLimitsIsEnforced() {
+    OpenSearchAdapter.TOTAL_FIELDS_LIMIT = 1;
```

Comment:
Not sure I'm super keen on overriding this - making the `TOTAL_FIELDS_LIMIT` a static would be better. For purposes of the test you can just for loop and try to create 5001 text strings, and then you wouldn't need to mutate or reset this value. That also has a benefit of catching any other potential issues with registering / storing 5k fields we may not be aware of.

Reply:
that is a good idea!